src/components/Button.svelte

```diff
@@ -0,0 +1,148 @@
+<script lang="ts">
+
+  import type { Snippet } from "svelte";
+  import { phosphorIcon } from "@data/icons";
+  import Icon from "@iconify/svelte";
+
+
+  interface Props {
+    children: Snippet;
+    style?: 'mauve' | 'peach' | 'blue' | 'green' | 'transparent';
+    small?: boolean;
+    phIconName?: string;
+    disabled?: boolean;
+  }
+
+  let { children, style, small, phIconName, disabled }: Props = $props();
+
+
+  let phIcon = $state({
```

Comment:
Shouldn't need a `$state` rune here.

Reply:
I would agree, but the intellisense kept yelling at me that if I'm going to change the value I should use `$state`. It runs with just a regular variable declaration as well tho.